### PR TITLE
Fish shell support

### PR DESCRIPTION
**Pull request by paysonwallach**
This PR adds support for Fish, resolving issue #14.

### CODE REVIEW

File: bashhub/shell/deps/fish/functions/__bh_check_bashhub_installation.fish [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Should only be sourced once following installation.
+#
+
+# Small function to check our Bashhub installation.
+function __bh_check_bashhub_installation
+    set -l ret 0
+    set -l config_path "$BH_HOME_DIRECTORY/config"
+    if [ -n "$FISH_VERSION" ] && [ -n "$__bh_enable_subshells" ] && [ "(trap)" -ne *"__bh_preexec_invoke_exec"* ]
+        echo "Bashhub's preexec hook is being overriden and is not saving commands. Please resolve what may be holding the DEBUG trap."
+        set ret 1
```
Comment (rcaloras) [inline bashhub/shell/deps/fish/functions/__bh_check_bashhub_installation.fish:12]: We can remove this section as its intended to check bash's trap being used for preexec

**Review by rcaloras — CHANGES_REQUESTED** [review]
@paysonwallach thanks for the PR! One small comment left. Can you also include an update to the Readme to include an install line for fish?

**Comment by rcaloras** [comment]
@paysonwallach any thoughts on these changes?

File: bashhub/shell/deps/fish/functions/__bh_check_bashhub_installation.fish (hunk 1 above)
Comment (paysonwallach) [reply bashhub/shell/deps/fish/functions/__bh_check_bashhub_installation.fish]: Ah, that makes sense. I've removed the check from that function.

**Comment by paysonwallach** [comment]
No trouble! I've updated the README with an installation instruction.

**Review by rcaloras — APPROVED** [review]

**Comment by rcaloras** [comment]
Dope! LGTM :+1: :tada:

**Comment by rcaloras** [comment]
@paysonwallach thanks for the contribution to support Fish! I'll cut a new release with this change.

**Comment by rcaloras** [comment]
@paysonwallach tried to install this on my local box for fish and having some issues with the install script. I'm getting an error: 
`install-bashhub.sh (line 174): Unexpected end of string, quotes are not balanced`

Should this work with fish version 3.0.2?


```
elementz@Kashmir:~/git/bashhub-client (master)$ fish 

Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

elementz@Kashmir ~/g/bashhub-client> echo "$FISH_VERSION" 
3.0.2
elementz@Kashmir ~/g/bashhub-client> fish install-bashhub.sh
install-bashhub.sh (line 174): Unexpected end of string, quotes are not balanced
(....file contents)
<W> fish: Error while reading file setup
```

**Pull request merged**